keyboards/tetris/keymaps/default/keymap.c

```diff
@@ -0,0 +1,747 @@
+}
+
+  switch ( keycode )
+  {
+/* 记录rgb 设置 */
+  case RGB_M_R:
+    save_rgbmode = true;
+    break;
+  case RGB_M_SW:
+    save_rgbmode = true;
+    break;
+  case RGB_M_X:
+    save_rgbmode = true;
+    break;
+  case RGB_M_G:
+    save_rgbmode = true;
+    break;
+
+  case RGB_MODE_KNIGHT:
+    save_rgbmode = true;
+    break;
+  case KC_00:
+  if (record->event.pressed) {
+        // Do something when pressed
+      } else {
+      register_code( KC_0 );
+      unregister_code( KC_0 );
+      register_code( KC_0 );
+      unregister_code( KC_0 );
+      }
+      return false; // Skip all further processing of this key
+
+case RESET:
+rgblight_mode( 1 );
+    rgblight_setrgb( 0, 0, 0 );
+    rgblight_setrgb_at(128, 0, 0,4);
+    rgblight_setrgb_at(6, 128, 24,36);
+    break;
+case RGB_TAP:
+if ( !(record->event.pressed))
+{
+if (RGB_TAP_STATE) 
+{
+RGB_TAP_STATE=false;
+rgblight_mode( 1 );
+rgblight_setrgb( RGB_TAP_Base_Color );
+rgblight_setrgb_at(RGB_TAP_Off_Color,16);
+}
+else
+{
+RGB_TAP_STATE=true;
+rgblight_mode( 1 );
+rgblight_setrgb( RGB_TAP_Base_Color );
+rgblight_setrgb_at(RGB_TAP_On_Color,9);
+}
+}
+break;
+
+  /* 特殊 shift */
+  case KC_ENT:
+    s_keycode  = KC_QUOT;
+    k_shifted  = &ques_shifted;
+    break;
+  case KC_UP:
+    s_keycode  = KC_SLSH;
+    k_shifted  = &ques_shifted;
+    break;
+  case KC_DOWN:
+    s_keycode  = KC_BSLS;
+    k_shifted  = &ques_shifted;
+    break;
+  default:
+    return(true);
+  }
+
+  shifted = get_mods() & (MOD_BIT( KC_LSHIFT ) | MOD_BIT( KC_RSHIFT ) );
+
+  /* Keydown. If shift is currently pressed, register its alternate keycode. */
+  if ( record->event.pressed && shifted )
+  {
+    *k_shifted = true;
+    register_code( s_keycode );
+    return(false);
+    /*
+     * Keyup. If shift was pressed back when the key was pressed, unregister
+     * its alternate keycode.
+     */
+  } else if ( !(record->event.pressed) && *k_shifted )
+  {
+    *k_shifted = false;
+    unregister_code( s_keycode );
+    return(false);
+    /* Otherwise, behave as normal. */
+  } else {
+    return(true);
+  }
+
+
+
+
+}
+
+
+void led_set_user( uint8_t usb_led )
+{
+  static uint8_t old_usb_led = 0;
+  _delay_ms( 10 ); /* gets rid of tick */
+
+  if ( (usb_led & (1 << USB_LED_CAPS_LOCK) ) && !(old_usb_led & (1 << USB_LED_CAPS_LOCK) ) )
+  {
+    /* If CAPS LK LED is turning on... */
+if (caps == false)
+{
+//RGB_LAYER0_mode = rgblight_config.mode;
+}
+caps= true;
+rgblight_setrgb( 0,0,0 );
+rgblight_setrgb_at(RGB_Caps_Color,12);
+#ifdef AUDIO_ENABLE
+    PLAY_SONG( tone_caps );
+#endif
+  }else if ( !(usb_led & (1 << USB_LED_CAPS_LOCK) ) && (old_usb_led & (1 << USB_LED_CAPS_LOCK) ) )
+  {
+    /* If CAPS LK LED is turning off... */
+caps= false;
+if (RGB_TAP_STATE==false){
+    rgblight_mode( RGB_LAYER0_mode );
+}else{
+rgblight_mode(1);
+rgblight_setrgb( 0,0,0 );
+}
+
+#ifdef AUDIO_ENABLE
+    PLAY_SONG( tone_caps );
+#endif
+  }
+
+  old_usb_led = usb_led;
+}
+
+#ifdef AUDIO_ENABLE
+
+void startup_user()
```

Comment:
This shouldn't be needed. It should play startup audio just fine, if it's defined correctly.

Reply:
As long as these are defined in your config.h, it will be picked up by the system:
https://docs.qmk.fm/#/feature_audio